addon/background.js

```diff
@@ -113,14 +113,24 @@ browser.webRequest.onBeforeSendHeaders.addListener(function (info) {
 // Remove X-Frame-Options to allow any page to be embedded in an iframe
 chrome.webRequest.onHeadersReceived.addListener(function (info) {
   let headers = info.responseHeaders;
+  let madeChanges = false;
   for (let i = 0; i < headers.length; i++) {
     let name = headers[i].name.toLowerCase();
     if (name === "x-frame-options" || name === "frame-options") {
       headers.splice(i, 1);
-      return {"responseHeaders": headers};
+      i--;
+      madeChanges = true;
+    }
+    if (name === "content-security-policy") {
+      headers[i].value = headers[i].value.replace(/frame-ancestors[^;]*/i, "");
```

Comment:
Just so I am sure I understand, this matches anything that starts with frame-ancestors, and then any characters up to a ;

Won't this leave the ; in the string? I guess that it is probably legal to have multiple ; in a row.

Reply:
It would, yeah. Since I have to fix a lint thing I'll just add `;?` here too